deeppavlov/models/sklearn/sklearn_component.py

```diff
@@ -0,0 +1,357 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Tuple, Any, Union, Generator, Iterable
+import numpy as np
+from scipy.sparse import spmatrix
+import pickle
+from pathlib import Path
+from scipy.sparse import issparse, csr_matrix
+from scipy.sparse import vstack, hstack
+import inspect
+
+from deeppavlov.core.common.errors import ConfigError
+from deeppavlov.core.common.registry import register, cls_from_str
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.models.estimator import Estimator
+
+log = get_logger(__name__)
+
+
+@register("sklearn_component")
+class SklearnComponent(Estimator):
+    """
+    Class implements wrapper for sklearn components for feature extraction,
+    feature selection, classification, regression etc.
+
+    Args:
+        model_name: string with full name of sklearn model to use, e.g. ``sklearn.linear_model:LogisticRegression``
+        save_path: save path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be saved to ``model_path/model.pkl``)
+        load_path: load path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be loaded from ``model_path/model.pkl``)
+        infer_method: string name of class method to use for infering model, \
+            e.g. ``predict``, ``predict_proba``, ``predict_log_proba``, ``transform``
+        kwargs: dictionary with parameters for the sklearn model
+
+    Attributes:
+        model: sklearn model instance
+        model_name: string with full name of sklearn model to use, e.g. ``sklearn.linear_model:LogisticRegression``
+        model_params: dictionary with parameters for the sklearn model without pipe parameters
+        pipe_params: dictionary with parameters for pipe: ``in``, ``out``, ``fit_on``, ``main``, ``name``
+        save_path: save path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be saved to ``model_path/model.pkl``)
+        load_path: load path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be loaded from ``model_path/model.pkl``)
+        infer_method: string name of class method to use for infering model, \
+            e.g. ``predict``, ``predict_proba``, ``predict_log_proba``, ``transform``
+        epochs_done: number of epochs done
+        batches_seen: number of batches seen
+        train_examples_seen: number of train examples seen
+    """
+    def __init__(self, model_name: str,
```

Comment:
`model_name` → `model_class`?

Reply:
I thought it's more clear that this parameter is a string with `name`. Maybe `class_name``?

Reply:
For components we have a `name` for registered name and `class` for this exact behavior.

Reply:
ok, `model_class`. fixed